tower-service/src/lib.rs

```diff
@@ -234,6 +236,79 @@ pub trait Service<Request> {
     /// Implementations are permitted to panic if `call` is invoked without
     /// obtaining `Poll::Ready(Ok(()))` from `poll_ready`.
     fn call(&mut self, req: Request) -> Self::Future;
+
+    /// Undo a successful call to `poll_ready`.
```

Comment:
Maybe I missed this but would be good to mention that most uses cases need to just forward the inner disarm.